lib/src/gotrue_client.dart

```diff
@@ -216,16 +238,53 @@ class GoTrueClient {
     return authResponse;
   }
 
-  /// Log in an existing user via a third-party provider.
+  /// Generates a link to log in an user via a third-party provider.
   Future<OAuthResponse> getOAuthSignInUrl({
     required Provider provider,
     String? redirectTo,
     String? scopes,
     Map<String, String>? queryParams,
   }) async {
     _removeSession();
-    return _handleProviderSignIn(provider,
-        redirectTo: redirectTo, scopes: scopes, queryParams: queryParams);
+    return _handleProviderSignIn(
+      provider,
+      redirectTo: redirectTo,
+      scopes: scopes,
+      queryParams: queryParams,
+    );
+  }
+
+  /// Verifies the PKCE code verifyer and retrieves a session.
+  Future<AuthResponse> exchangeCodeForSession(String authCode) async {
+    assert(_asyncStorage != null,
+        'You need to provide asyncStorage to perform pkce flow.');
+
+    final codeVerifier = await _asyncStorage!
+        .getItem(key: '${Constants.defaultStorageKey}-code-verifier');
+
+    final Map<String, dynamic> response = await _fetch.request(
+      '$_url/token',
+      RequestMethodType.post,
+      options: GotrueRequestOptions(headers: _headers, body: {
+        'auth_code': authCode,
+        'code_verifier': codeVerifier,
+      }, query: {
+        'grant_type': 'pkce',
+      }),
```

Comment:
```suggestion
      },),
```
Makes the formatting much better 😅